examples/recognition_pipeline/recognition_pipeline.py

```diff
@@ -0,0 +1,198 @@
+The labeling content is updated in the database for each image.
+By the end of the process, the project's admin group get notified.
+"""
+
+from __future__ import absolute_import
+
+import argparse
+import logging
+import random
+from datetime import datetime
+
+from past.builtins import unicode
+
+from firebase_admin import firestore
+
+import apache_beam as beam
+from apache_beam.io import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+
+from providers import google_vision_api
+from pipeline_lib.image_filtering import is_eligible
+from pipeline_lib.redefine_labels import RedefineLabels
+
+RANGE_OF_BATCH = 10 # Defines the range of the each batch for querying the database.
+collection_name = 'Images'
+
+def initialize_DB(): 
+    """Initializes project's Firestore databse for writing and reading purposes.
+
+    """
+    if not firebase_admin._apps:
+        firebase_admin.initialize_app(credentials.ApplicationDefault(), {
+        'projectId': 'step-project-ellispis',
+        })
+    return firestore.client()
+
+def get_provider(provider_name):
+    """ Creates an object of type ImageRecognitionProvider by the specific provider input.
+
+    """ 
+    if provider_name == 'Google_Vision_API':
+        return google_vision_api.GoogleVisionAPI()
+    
+class GetDataset(beam.DoFn):
+    """Queries the project's database to get the image dataset to label.
+
+    """
+    def setup(self):
+        self.db = initialize_DB()
+
+    def process(self, element, ingestion_provider = None, ingestion_run = None):
+        """Queries firestore database for images from the ingestion_provider within a random range (by batch).
+        
+        Args:
+            element: the lower limit for querying the database by the random field.
+            ingestion_provider: the input of the pipeline, determines the images dataset.
+            ingestion_run: the input of the pipeline, determines the dataset.
+
+        Returns:
+            A list of dictionaries with all the information (fields and id) of each one of the Firestore query's image documents.
+        """
+        random_min = element
+        random_max = element+RANGE_OF_BATCH-1 # the higher limit for querying the database by the random field.
+        # TODO: have different queries for ingestion provider and ingestion run once relevant data has been uploaded to Firestore by ingestion pipeline.
+        if ingestion_provider:
+            query = self.db.collection(collection_name).where(u'attribution', u'==', ingestion_provider).where(u'random', u'>=', random_min).where(u'random', u'<=', random_max).stream()
+        else:
+            query = self.db.collection(collection_name).where(u'attribution', u'==', ingestion_run).where(u'random', u'>=', random_min).where(u'random', u'<=', random_max).stream()
```

Comment:
what if both are provided? Then still `ingestion_provider` is used!
Turn it around or check that only one of them is set

Reply:
changed lke we talked in chat.